### PR TITLE
Update docs on data structures

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: tutorial/patterns.md [hunk 1]
```diff
@@ -115,27 +115,25 @@ A getter should yield the same value every time.
...
 ## Data Structures
 
-Prefer to use Vector over JavaScript `Array` for internal use for its extra flexibility.
 
 For public API, we have three options: JavaScript `Array`, 
-`IIterator`, and `ISequence`.
+`IIterator`, and `ReadonlyArray` (an interface defined by TypeScript).
 
 Prefer an `Array` for:
+- A and that is meant to be mutable.
```
Comment (sccolbert) [inline tutorial/patterns.md]: s/A and/An array?

File: tutorial/patterns.md [hunk 2]
```diff
@@ -115,27 +115,25 @@ A getter should yield the same value every time.
...
+
+Prefer an a `ReadonlyArray`
```
Comment (sccolbert) [inline tutorial/patterns.md]: s/an a/a

File: tutorial/patterns.md [hunk 3]
```diff
@@ -115,27 +115,25 @@ A getter should yield the same value every time.
...
-- A signal payload.
-- A public attribute that is inherently static.  Use `.slice()` to
-make sure the internal value cannot be mutated by the consumer.
+- A signal payload - since it will be consumed by multiple listeners.
+- The values may need to be accessed randomly.
+- A public attribute that is inherently static.
 
 Prefer an `IIterator` for:
-- A return value where the value is based on an internal `Vector` but the 
+- A return value where the value is based on an internal `Array` but the
```
Comment (sccolbert) [inline tutorial/patterns.md]: s/internal Array/internal data structure?

**Comment by blink1073** [comment]
Comments addressed.

**Pull request merged**